check-licenses/action.yml

```diff
@@ -0,0 +1,82 @@
+name: "Check licenses"
+description: "Check if package licenses in the current environment are among the accepted ones"
+
+inputs:
+  python-version:
+    description: 'Desired Python version.'
+    default: '3.10'
+    required: false
+    type: string
+  use-python-cache:
+    description: 'Whether to use Python cache or not. Requires a pyproject.toml file.'
+    required: false
+    default: true
+    type: boolean
+  skip-install:
+    description: 'Skip project installation.'
+    default: false
+    required: false
+    type: boolean
+  target:
+    description: 'Optional target to be passed for the build process.'
+    required: false
+    default: ''
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Git and clone project"
+      if: inputs.skip-install == 'false'
+      uses: actions/checkout@v3
+
+    - name: "Set up Python"
+      if: inputs.skip-install == 'false'
+      uses: pyansys/actions/_setup-python@main
+      with:
+        python-version: ${{ inputs.python-version }}
+        use-cache: ${{ inputs.use-python-cache }}
+
+    - name: "Update pip"
+      if: inputs.skip-install == 'false'
+      shell: bash
+      run: python -m pip install -U pip
+
+    - name: "Check if specific target is requested"
+      shell: bash
+      run: |
+        echo "install_target=$( [[ '${{ inputs.target }}' == '' ]] && echo '.' || echo '.[${{ inputs.target }}]')" >> $GITHUB_ENV
+
+    - name: "Install Python library"
+      if: inputs.skip-install == 'false'
+      shell: bash
+      run: pip install ${{ env.install_target }}
+
+    - name: "Download txt files so they are found from other repos"
+      shell: bash
+      run: |
+        wget https://raw.githubusercontent.com/pyansys/actions/feat/check-licenses/check-licenses/accepted-licenses.txt
+        wget https://raw.githubusercontent.com/pyansys/actions/feat/check-licenses/check-licenses/ignored-packages.txt
```

Comment:
Are you sure that the moment we merge this PR (and the branch gets deleted), those links will still work? I'd say they won't... 
```suggestion
    - name: "Download txt files so they are found from other repos"
      shell: bash
      run: |
        wget https://raw.githubusercontent.com/pyansys/actions/main/check-licenses/check-licenses/accepted-licenses.txt
        wget https://raw.githubusercontent.com/pyansys/actions/main/check-licenses/ignored-packages.txt
```

You'll probably have to merge this PR with a failing status due to this breaking change. But once it's on main, it should work.

In fact, this link will have to be edited every time we do a release so that it points to whatever release branch we have... It ain't gonna be easy to maintain... We should think something... 

What about using this action to do that same operation? https://github.com/marketplace/actions/checkout-files

Reply:
I have a question here that I was discussing with Maxime on Friday.

Won't it work if we leave this as feat/check-licenses for now? It will still point to the feat branch, but this branch will still be available until it gets removed right? I was thinking about first merging and then opening another PR to change from feat/check-licenses to main. Am I missing something?

I really like the approach you suggested with the checkout files action. I am going to implement it now.

Reply:
Sure, we can do it in 2 PRs =) there is no issue in that. I was just trying to optimize. But at least we will get our green ticks dopamine 😄 

And yeah, let's try to use the action. I think it'll be more clear for everybody to understand what this step is doing. I know what you're doing, and probably anybody that understand what ``wget`` is knows what you're doing... but just in case =)

Reply:
For now, I'll leave it like this since I want to test the action from the PR in ``pyansys/pyansys``. After checking that everything works, I'll modify all the references to @feat/check-licenses to @main.

Reply:
@RobPasMue. I have implemented the [action](https://github.com/marketplace/actions/checkout-files) following the documentation. However, when using the ``check-licenses`` action in the [pyansys/pyansys PR](https://github.com/pyansys/pyansys/pull/97) it looks like it is trying to find the files in the ``pyansys/pyansys`` repo, see url that is trying to access:

![image](https://user-images.githubusercontent.com/53979143/204278499-465ed1fc-9265-43b9-82aa-4ba15d165233.png)

I don't see an option to define the repo in which the files are located... Is it possible that this action is only capable of checking out files that are in the same repo where the action is called?

Reply:
I think there shouldn't be a problem to define these files in the `main` branch, even if those point from a release branch. They are expected to be quite stable.

Reply:
Ha true... it looks like this action is only valid for files in the same repo... sorry for that =( you might need to revert to the ``wget`` mechanism

Reply:
Summarizing:
* Let's revert to the ``wget`` mechanism
* Point directly to its expected location in the ``main`` branch
* Merge whenever you feel it's ready =)